pyomo/contrib/doe/result.py

```diff
@@ -114,15 +112,21 @@ def calculate_FIM(self, dv_values, result=None):
         # get number of parameters
         no_param = len(self.para_name)
 
-        # reform jacobian, split the overall Q into Q_r, each r is a flattened measurement name
-        Q_response_list, variance_list = self._jac_reform_3D(
-            self.jaco_information, Q_response=True
-        )
-
         fim = np.zeros((no_param, no_param))
 
-        for i in range(len(Q_response_list)):
-            fim += (1 / variance_list[i]) * (Q_response_list[i] @ Q_response_list[i].T)
+        # convert dictionary to a numpy array
+        Q_all = []
+        for par in self.para_name:
+            Q_all.append(self.jaco_information[par])
+        n = len(self.para_name)
+
+        Q_all = np.asarray(Q_all).T
```

Comment:
Replace this with Q_all = np.array(list(self.jaco_information[p] for p in self.para_name)).T